lib/formtastic/inputs/check_boxes_input.rb

```diff
@@ -180,11 +180,18 @@ def checkbox_input(choice)
       def make_selected_values
         if object.respond_to?(method)
           selected_items = object.send(method)
-
           # Construct an array from the return value, regardless of the return type
           selected_items = [*selected_items].compact.flatten
 
-          [*selected_items.map { |o| send_or_call_or_object(value_method, o) }].compact
+          selected = []
+          selected_items.each do |o|
+            item = send_or_call_or_object(value_method, o)
+            if item == nil
+              item = o.id if o.respond_to? :id
+            end
+            selected << item
+          end
+          selected
```

Comment:
And I assume we still need the `.compact` call at the end, to filter out `nil`s?